### DISCUSSION TITLE
"Resize and adjust Height/Width" resize mode for img2img

### DISCUSSION THREAD

**Discussion by Kinexity** [hlky/stable-diffusion-webui#655] [Feature suggestions]
I'd like to propose resize mode for img2img where image gets resized to fit within given height and width without changing aspect ratio just like with "Resize and fill" but without filling but instead automatically adjusting height or width instead. To give an example: let's say we want to generate 512x512 image but the input image is 1024x512. The image, by my suggestion, would get resized to 512x256 and the generated image would also be 512x256.

Edit: Maybe I didn't make myself clear. I want new resize mode to preserve aspect ratio of the original image in the output image and for it to automatically choose height and width of generated image.

**Comment by thedyze** [comment]
I really don't follow the logic here, you want to generate a 512x512 image, but you want it to become 512x256?
If the input image is 1024x512, just set the output to 512x256 and everything will be fine?

**Reply by Kinexity** [reply]
The logic stems from the fact that under lower denoising setting (as in ~0.5) the filling added by resize and fill produces garbage that I'd cut out anyways. The point is to not spend time and compute generating this garbage. The example I gave was slightly extreme because normally you'd set generation height and width to aspect ratio closer to the original image.

Edit: Maybe I didn't make myself clear. I want new resize mode to preserve aspect ratio of the original image in the output image and for it to automatically choose height and width of generated image.
